### DISCUSSION TITLE
Mixed spinful fermion-spin 1/2 basis

### DISCUSSION THREAD

**Discussion by bigm4t** [QuSpin/QuSpin#683] [Q&A]
Dear all,

I'm struggling to define a Hamiltonian in a spinful fermion-spin 1/2 mixed basis.
I provide a minimal example (one site) below:

```
import numpy as np
from quspin.operators import hamiltonian # Hamiltonians and operators
from quspin.basis import spinful_fermion_basis_1d, spinless_fermion_basis_1d, spin_basis_1d, tensor_basis

fermion_basis=spinful_fermion_basis_1d(1)
spin_basis = spin_basis_1d(1, pauli=0)
fs_basis = tensor_basis(fermion_basis, spin_basis)

J = 1.0 
j_up = [[J,0,0,0]]
static=[['+-||z',j_up]]
no_checks = dict(check_pcon=False,check_symm=False,check_herm=True)
H=hamiltonian(static,[],basis=fs_basis,dtype=np.float64,**no_checks)
```

Creating the basis with `tensor_basis()` works all fine, but calling the `hamiltonian()` function returns
`ValueError: missing '|' charactor in: +-, (0, 0)`

Interestingly, defining the spinful basis by combining two spinless basis doesn't yield an error (see code below). However, this is not a valid workaround because I need the fermionic objects to anticommute (and this approach treats the fermions coming from different basis as commuting objects). 

```
import numpy as np
from quspin.operators import hamiltonian # Hamiltonians and operators
from quspin.basis import spinful_fermion_basis_1d, spinless_fermion_basis_1d, spin_basis_1d, tensor_basis

fermion_basis=spinless_fermion_basis_1d(1)
spin_basis = spin_basis_1d(1, pauli=0)
fs_basis = tensor_basis(fermion_basis, fermion_basis, spin_basis)

J = 1.0 
j_up = [[J,0,0,0]]
static=[['+-||z',j_up]]
no_checks = dict(check_pcon=False,check_symm=False,check_herm=True)
H=hamiltonian(static,[],basis=fs_basis,dtype=np.float64,**no_checks)
```

Any help would be much appreciated!

**Comment by bigm4t** [comment]
It seems that inverting the order in the `tensor_basis()` definition works:

```
import numpy as np
from quspin.operators import hamiltonian # Hamiltonians and operators
from quspin.basis import spinful_fermion_basis_1d, spinless_fermion_basis_1d, spin_basis_1d, tensor_basis

fermion_basis=spinful_fermion_basis_1d(1)
spin_basis = spin_basis_1d(1, pauli=0)
fs_basis = tensor_basis(spin_basis,fermion_basis)

J = 1.0
j_up = [[J,0,0,0]]

static=[['z|+-|',j_up]]

no_checks = dict(check_pcon=False,check_symm=False,check_herm=True)
H=hamiltonian(static,[],basis=fs_basis,dtype=np.float64,**no_checks)
```

I'll leave the question open in case someone can answer why the first code snippet wasn't working.

Cheers

**Comment by SS007-M** [comment]
Hi!, if we create a tensor basis for spin 1/2- spin 1 (1D system) ; any idea how to write the Heisenberg Hamiltonian correctly in the mixed-spin case?
